R/plot_xctr_portfolio_financial.R

```diff
@@ -0,0 +1,52 @@
+#' Plot xctr data on a portfolio level, with financial data
+#'
+#' @param data A data frame like [financial].
+#' @param mode A string. Name of the financial weight.
+#'
+#' @return A [ggplot] object.
+#' @export
+#'
+#' @examples
+#' plot_xctr_portfolio_financial(financial, mode = "worst_case")
+plot_xctr_portfolio_financial <- function(data, mode = c("equal_weight", "worst_case", "best_case", "main_activity")) {
```

Comment:
Good point ! I'll create another ticket for that, since the logic now for financial and with no financial data is to have two separate functions for each level : 

- plot_xctr_portfolio()
-plot_xctr_company()
-plot_xctr_portfolio_financial()
-plot_xctr_portfolio_company()

Also this logic was brought up [here](https://github.com/2DegreesInvesting/tiltPlot/pull/21#discussion_r1265856470) 

But I agree that it would be great to regroup the common code into one - I'll also sync with Anne about that :) 
Thanks @maurolepore !